data/tutorials/getting-started/02_fst_program.md

```diff
@@ -0,0 +1,310 @@
+- Modules are bundles of named values 
+- Identical names from distinct modules don't clash
+- The standard library is a bunch of modules
+
+In addition to that, modules are also the mean of organising projects. Concerns are separated into isolated modules. This is outlined in the next section. Before that, let's see how to use a definition from a module in our project: 
+```shell
+$ sed -i 's/print_endline/Printf.printf "%s!\\n"/g' bin/main.ml
+```
+
+This replaces the function `print_endline` with the function `printf` from the `Printf` module in the standard library. Building and executing this modified version should produce almost the same output as before. Use `dune exec hello` to try it for yourself.
+
+## Every File Defines a Module
+
+Most importantly, each OCaml file defines a module, once compiled. This is how separate compilation works in OCaml. Each sufficiently standalone concern should be isolated into a file-based module. References to external modules create dependencies. Circular dependencies between modules are not allowed.
+
+To create a module, let's create a file named `lib/hello.ml`:
+```shell
+$ echo 'let world = "Hello from a module"' > lib/hello.ml
+```
+
+Here is a new version of the `bin/main.ml` file and execution of the resulting project:
+```shell
+$ echo 'let () = Printf.printf "%s\n" Hello.world' > bin/main.ml
+
+$ dune exec hello
+Entering directory '/home/cuihtlauac/caml/ocaml.org'
+Hello from a module
+```
+
+The file `lib/hello.ml` contains the module named `Hello`, which in turn defines a string named `world`. This definition is referred to as `Hello.world` from the `bin/main.ml` file.
+
+Dune can launch UTop to access to the modules exposed by a project interactively. Here's how:
+```shell 
+$ dune utop
+```
+
+Then, inside the `utop` toplevel, it is possible to inspect our `Hello` module, just like we did when we examined the `Option` module in the previous tutorial.
+```ocaml
+# #show Hello;;
+module Hello : sig val world : string end
+```
+
+## Defining Module Interfaces
+
+What UTop's `#show` command displays is an API, the list of definitions provided by a module, which is called a _module interface_. In a similar way, module can be defined by `.ml` files. It is also possible to define module interfaces from files. Such files have a `.mli` extension and must share the same name part. 
+```shell
+$ echo 'val world : string' > lib/hello.mli
+```
+
+Observe that only what is between `sig` and `end` needs to written in the interface file. This is explained in the tutorial dedicated to [modules](link_modules).
+
+Module interfaces are also used to create _private_ definitions. A module definition is private if it is not listed in its corresponding interface. If no interface file exists, everyting is public.
+
+Edit the `lib/hello.ml` file to add the `mundo` definition, as shown below:
+```ocaml
+let mundo = "¡Hola Mundo!"
+let world = mundo
+```
```

Comment:
Are we *adding* the these lines in `lib/hello.ml` or *replacing* what's there?
I got errors when I added it:

```
1 | let world = "Hello from a module"
        ^^^^^
Error (warning 32 [unused-value-declaration]): unused value world.


```

Reply:
(Replace)